File: content/blog/two-sum/go/bruteforce.go
package main

func twoSumBruteforce(nums []int, target int) []int {
	for firstIdx, main := range nums {
		for secondIdx, complement := range nums {
			if firstIdx == secondIdx {
				continue
			}
			if main+complement == target {
				return []int{firstIdx, secondIdx}
			}
		}
	}

	return nil
}

File: content/blog/two-sum/ts/bruteforce.ts
export const twoSumBruteForce = (nums: number[], target: number): number[] => {
    for (let i = 0; i < nums.length; i++) {
        for (let j = 0; j < nums.length; j++) {
            if (i !== j && nums[i] + nums[j] === target) {
                return [i, j]
            }
        }
    }

    return []
}